zip-0317.rst

```diff
@@ -262,75 +262,86 @@ following section is planned to be implemented by `zcashd` and `zebrad`.
 Recommended algorithm for block template construction
 '''''''''''''''''''''''''''''''''''''''''''''''''''''
 
-Define a constant :math:`weight\_cap = 4`.
+Define constants :math:`weight\_ratio\_cap = 4` and
+:math:`block\_unpaid\_action\_limit = 50\!`.
 
 Let :math:`conventional\_fee(tx)` be the conventional fee for transaction
 :math:`tx` calculated according to the section `Fee calculation`_.
 
+Let :math:`unpaid\_actions(tx) = \mathsf{max}\!\left(0,\, \mathsf{max}(grace\_actions,\, tx.\!logical\_actions) - \mathsf{floor}\!\left(\frac{tx.fee}{marginal\_fee}\right)\right)\!`.
+
+Let :math:`block\_unpaid\_actions(block) = \sum_{tx \,\in\, block}\, unpaid\_actions(tx)`.
+
```

Comment:
Zebra's initial implementation forgot to subtract the coinbase transaction from the block sigop and size limits. So let's be a bit more explicit about it.

```suggestion

Let :math:`B` be the block template. Initially, :math:`B` only includes the coinbase transaction.

```